core/authentication_api.php

```diff
@@ -764,29 +783,29 @@ function auth_does_password_match( $p_user_id, $p_test_password ) {
  * @access public
  */
 function auth_process_plain_password( $p_password, $p_salt = null, $p_method = null ) {
-	$t_login_method = config_get( 'login_method' );
-	if( $p_method !== null ) {
-		$t_login_method = $p_method;
-	}
+	$t_login_method = $p_method !== null ? $p_method : config_get( 'login_method' );
 
 	switch( $t_login_method ) {
 		case CRYPT:
-
 			# a null salt is the same as no salt, which causes a salt to be generated
 			# otherwise, use the salt given
 			$t_processed_password = crypt( $p_password, $p_salt );
 			break;
 		case MD5:
 			$t_processed_password = md5( $p_password );
 			break;
+		case SAFE_HASH:
+			$t_processed_password = password_hash( $p_password, PASSWORD_DEFAULT );
+			break;
 		case BASIC_AUTH:
 		case PLAIN:
 		default:
 			$t_processed_password = $p_password;
 			break;
 	}
 
-	# cut this off to DB_FIELD_SIZE_PASSWORD characters which the largest possible string in the database
+	# cut this off to DB_FIELD_SIZE_PASSWORD characters which is the largest
```

Comment:
I would error instead of truncate when setting a password that exceeds db schema size.